### PR TITLE
Add and show plants

### LINKED ISSUE #13: Basic Add Plants flow
see #8

### PR DESCRIPTION
**Pull request by mheiber**
fix #13 

## Description

This flow now works:

- Empty home screen just shows a button. Tap it.
- choose a contact
- choose a plant kind (cactus, circle tree, etc.)
- choose a watering frequency
- see all of the plants you've added on the Home screen.

There's no persistence or styling yet.

## Testing

There's 100% test coverage for .ts files, but I haven't done any integration-style tests yet, which would exercise the screens.

There are a lot of test snapshots, which make the diff seem big.

This PR involves many tweaks to config to get Jest to play nicely with our React Native libraries.

## iOS Screenshots

Screenshot 2020-12-26 at 22 18 52
Screenshot 2020-12-26 at 22 18 58
Screenshot 2020-12-26 at 22 19 04
Screenshot 2020-12-26 at 22 19 15

## Android Screenshots

home-after-adding
choice

choice-kind

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,39 +6,78 @@
...
     "@types/jest": "^25.2.3",
     "@types/react-native": "^0.63.2",
     "@types/react-test-renderer": "^16.9.2",
+    "@types/uuid": "^8.3.0",
     "@typescript-eslint/eslint-plugin": "^2.27.0",
     "@typescript-eslint/parser": "^2.27.0",
     "babel-jest": "^25.1.0",
     "eslint": "^6.5.1",
     "jest": "^25.1.0",
+    "jest-transform-stub": "^2.0.0",
```
Comment (mmxw) [inline package.json]: are this and `react-test-renderer` and `react-dom` still used?

File: package.json [hunk 2]
```diff
@@ -6,39 +6,78 @@
...
+      }
+    },
     "moduleFileExtensions": [
       "ts",
       "tsx",
       "js",
       "jsx",
       "json",
       "node"
+    ],
```
Comment (mmxw) [inline package.json]: what's this for?

File: src/App.test.tsx [hunk 3]
```diff
@@ -4,11 +4,14 @@
...
 // Note: test renderer must be required after react-native.
-import renderer from 'react-test-renderer';
+import renderer, {act} from 'react-test-renderer';
 
-it('renders correctly', () => {
-  renderer.create(<App />);
+it('renders correctly', async () => {
+  // needed because react-navigation does async stuff
+  await act(async () => {
+    renderer.create(<App />);
```
Comment (mmxw) [inline src/App.test.tsx]: if we use react testing library as we did in other places, do we need the double `async`s here?

File: src/home-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  return (
+    <View>
+      <Plants plants={plants} />
+      <Button onPress={pickContact} title="pick" />
+    </View>
+  );
+}
+
+export const exportedForTesting = {
```
Comment (mmxw) [inline src/home-screen.tsx]: i don't seem to be able to find where this is used. what is this for?

File: README.md [hunk 5]
```diff
@@ -21,6 +21,18 @@ or
...
+Watch mode:
+
+```sh
+yarn test --watch
+```
+
+To update snapshots:
+
+```sh
+yarn jest --update-snapshot
```
Comment (mmxw) [inline README.md]: typo? `jest` should be `test`? 
```suggestion
yarn test --update-snapshot
```

File: src/plant-choice-screen.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+      <WateringPicker
+        contactName={contact.name}
+        onPick={handlePickWaterFrequency}
+      />
+    );
+  }
+
+  return (
+    <View>
+      <Text>Pick an appearance for {contact.name}</Text>
```
Comment (mmxw) [inline src/plant-choice-screen.tsx]: `appearance` seems a bit confusing to me. maybe use `species`?

File: package.json (hunk 2 above)
Comment (mheiber) [reply package.json]: It was stray.

File: src/App.test.tsx (hunk 3 above)
Comment (mheiber) [reply src/App.test.tsx]: Nope, removed.

We can use `render` from `@testing-library/react-native`, but need to do something async (such as `findAllByA11yLabel`) in order to not get an error message about `act` (https://reactjs.org/docs/test-utils.html#act).

I made the change.

File: src/home-screen.tsx (hunk 4 above)
Comment (mheiber) [reply src/home-screen.tsx]: stray - sorry about that

File: package.json (hunk 1 above)
Comment (mheiber) [reply package.json]: - `jest-transform-stub` was unused - good catch, I removed it
- `react-test-renderer` is a peer dependency of `@testing-library/react-native`
- `react-dom` is an undeclared dependency of recoil, if we remove it we get `Cannot find module 'react-dom' from 'recoil.js'` when running tests 🤦‍♂️. I commented on a similar issue https://github.com/facebookexperimental/Recoil/issues/755#issuecomment-751525879

**Review by mmxw — APPROVED** [review]

**Pull request merged**